src/schema/organization/contributors/resolvers.js

```diff
@@ -1,61 +1,131 @@
 import fetchData from '../../../github/dataFetcher';
 import githubQueries from '../../../github/queries';
-import { GeneralSet, getOldestDate } from '../../../utils'
+import { getOldestDate } from '../../../utils';
+
+/**
+ * @class A variant of Set, which the difference is on the equality condition of objects
+ */
+export class ContributorSet {
+  constructor() {
+      this.map = new Map();
+      this[Symbol.iterator] = this.values;
+  }
+
+  add(item) {
+    this.map.set(item.login, item);
+  }
+
+  values() {
+      return this.map.values();
+  }
+
+  delete(item) {
+      return this.map.delete(item.login);
+  }
+}
 
 /**
  * Process the repositories array to leave only the information the schema needs.
  * @param {Array} repositories 
  * @return processed repositories array
  */
-const process = (repositories) => {
-  const contributors = new GeneralSet();
+const processContributors = (repositories) => {
+  const contributors = new ContributorSet();
 
   repositories.forEach(repository => {
     repository.mentionableUsers.nodes.forEach(user => {
-      const firstPullRequest = user.contributionsCollection.pullRequestContributions.nodes[0];
-      const firstIssue = user.contributionsCollection.issueContributions.nodes[0];
-      const firstReview = user.contributionsCollection.pullRequestReviewContributions.nodes[0];
-
-      let currentDate = undefined;
-
-      if (firstPullRequest || firstIssue || firstReview) {
-        const firstContributions = [firstPullRequest, firstIssue, firstReview];
-        const dates = [];
-        
-        firstContributions.forEach(contribution => {
-          if (contribution) dates.push(contribution.firstContributionDate);
-        });
-
-        currentDate = getOldestDate(dates);
-      }
-
-      user.contributionsCollection = undefined;
-      user.firstContributionDate = currentDate;
+      user.a = 1;
```

Comment:
what is `a`

Reply:
Debugging garbage. I will remove. Thanks